src/analysis/thermo/thermoFBA/addLoopLawConstraints.m

```diff
@@ -259,7 +332,77 @@
 else
     display('method not found')
     method
-    pause;
+end
 end
 
+function conComp = connectedRxnsInNullSpace(N)
+% find connected components for reactions in cycles given a minimal feasible
+% nullspace as defined in Chan et al., 2017. Loopless constraints are required only 
+% for the connected components involving reactions required to have no flux 
+% through cycles (the target set) in the resultant flux distribution
+% Reactions in the same connected component have the same conComp(j).
+% Reactions not in any cycles, thus not in any connected components have conComp(j) = 0.
+conComp = zeros(size(N, 1), 1);
+nCon = 0;
+vCur = false(size(N, 1), 1);
+while any(conComp == 0 & any(N, 2))
+    vCur(:) = false;
+    % find the first reaction not in any connected component yet
+    vCur(find(conComp == 0 & any(N, 2), 1)) = true;
+    nCon = nCon + 1;
+    nCur = 0;
+    % loop until no new reaction is added
+    while nCur < sum(vCur)
+        nCur = sum(vCur);
+        % get any reactions sharing the same columns in the current component
+        vCur(any(N(:, any(N(vCur, :), 1)), 2)) = true;
+    end
+    conComp(vCur) = nCon;
+end
 end
+
+function rxnLink = getRxnLink(model, conComp, rxnInLoops)

```

Comment:
May you split out these functions into separate ones?

Reply:
todo